src/lib.rs

```diff
@@ -1,58 +1,68 @@
 /*!
-Input manager for [Bevy](https://bevyengine.org), inspired by
-[Unreal Engine Enhanced Input](https://dev.epicgames.com/documentation/en-us/unreal-engine/enhanced-input-in-unreal-engine).
+A powerful observer-based input manager for Bevy.
 
-# Quick start
+The design of this crate is heavily inspired by
+[Unreal Engine Enhanced Input](https://dev.epicgames.com/documentation/en-us/unreal-engine/enhanced-input-in-unreal-engine),
+but adapted to Bevy's ECS architecture and idioms.
 
-## Prelude
+# Core Concepts
 
-We provide a [`prelude`] module, which exports most of the typically used traits and types.
+This crate introduces three main concepts:
 
-## Plugins
+- **Actions** represent something a player can do, like "Jump", "Movement", or "Open Menu". They are not tied to specific input.
+- **Bindings** connect those actions to real input sources such as keyboard keys, mouse buttons, gamepad axes, etc.
+- **Contexts** represent a certain input state the player can be in, such as "On foot" or "In car". They associate actions with
+  entities and define when those actions are evaluated.
 
-Add [`EnhancedInputPlugin`] to your app:
+In short, actions are mapped to inputs via bindings, and contexts control which actions are active.
 
-```
-use bevy::prelude::*;
-use bevy_enhanced_input::prelude::*;
+## Actions
 
-let mut app = App::new();
-app.add_plugins((MinimalPlugins, EnhancedInputPlugin));
-```
+Each action represents a different in-game behavior.
+To create a new action, you will need to define a new struct and implement the [`InputAction`] trait for it,
+typically using the provided [`InputAction`] derive macro.
 
-## Core Concepts
+The action's output type is defined by the [`InputAction::Output`] associated type,
```

Comment:
Maybe mention that actions consume the binding by default? This important for context ordering.